tests/reftests/tree.test

```diff
@@ -0,0 +1,186 @@
+N0REP0
+### <pkg:a.1>
+opam-version: "2.0"
+### <pkg:b.1>
+opam-version: "2.0"
+depends: "a"
+### <pkg:c.1>
+opam-version: "2.0"
+depends: "b"
+### <pkg:d.1>
+opam-version: "2.0"
+### OPAMYES=1
+### # should fail with a no switch error
+### opam tree a
+[ERROR] No switch is currently set. Please use 'opam switch' to set or install a switch
+# Return code 50 #
+### opam switch create tree --empty
+### opam tree
+### opam tree a
+[WARNING] Not installed package a, skipping
+[ERROR] No package found
+# Return code 5 #
```

Comment:
This seems a bit surprising. Is there a reason why it can’t be done for non-installed packages?
I feel like users would want to have this feature before installing it to understand why a package requires another.

Reply:
I didn't support non-installed packages because the similar commands of other package managers don't support that.

> https://github.com/ocaml/opam/pull/5171#issuecomment-1175866838
> All of `cargo tree`, `npm ls`, and `yarn why`, which are mentioned in https://github.com/ocaml/opam/issues/3775 and inspired the current implementation, only take account of installed packages. So I guess it is ok to support only the installed packages.

I also thought it's hard to do, but I'll try it again as I now think it's possible (by using `OpamSolution.simulate_new_state`).

Reply:
Tried to address this in https://github.com/ocaml/opam/pull/5171/commits/c8db478465b82f438e4960140ff4502fa821a9d6. What do you think about it?